### DISCUSSION TITLE
FastAPI CLI not being able to understand the project structure

### DISCUSSION THREAD

**Discussion by SergioPereo** [fastapi/fastapi#12132] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
https://gist.github.com/SergioPereo/07bdbd545f343142856ad96678208927
```

### Description

I expect the FastAPI CLI to understand the full project structure. But seems like it cannot detect the twilio folder even though it have a __init__.py file

### Operating System

Linux

### Operating System Details

It uses python 3.12 docker so it is a debian:12, 12.7, bookworm

### FastAPI Version

fastapi==0.112.2 fastapi-cli==0.0.5

### Pydantic Version

pydantic==2.8.2

### Python Version

Python 3.12.4

**Comment by sinisaos** [comment]
@SergioPereo Your gist error logs say that the `schemas` module is missing and your project `tree` confirms that. I think you should add `schemas.py` to the `twilio` directory (maybe another name will be better because it's confusing with the `twilio-python` you imported from `site-packages`) and import it in `crud.py` like this
```python
from .schemas import MessageRequest, MessageResponse, UserIdFromPhone
```
and everything should be fine. This is how the new project structure `tree` should be look like. 
```bash
.
└── app
    ├── database.py
    ├── __init__.py
    ├── main.py
    └── twilio
        ├── crud.py
        ├── __init__.py
        ├── schemas.py
        └── users.py
```

Sorry if I miss your point.

**Comment by SergioPereo** [comment]
Oh yeah, sorry I forget to add schemas.py to the project structure in the gist but it is actually in the code. In fact I meant to say that the fastapi cli cannot see any module even if it has a __init__.py in it. The code above works when I run my Dockerfile like this:

```
FROM python:3.12

WORKDIR /code

RUN apt-get update \
    && apt-get -y install libpq-dev gcc \
    && pip install psycopg2

COPY ./requirements.txt /code/requirements.txt

RUN pip install --no-cache-dir --upgrade -r /code/requirements.txt

COPY ./app /code/app

CMD ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "8000"]
```

Instead of with the FastAPI cli command

The error was mine for not be totally clear, I apologize for that.

**Comment by sinisaos** [comment]
@SergioPereo I don't know exactly what is causing your problem, but in my case everything works fine with both 
`CMD ["fastapi", "run", "app/main.py"]` or `CMD ["uvicorn", "app.main:app", "--host", "0.0.0.0", "--port", "8000"]`

[cli.webm](https://github.com/user-attachments/assets/f5f3b394-7dbb-4ca3-be7f-a0da4811b8b2)

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
